lib/templ.ml

```diff
@@ -245,29 +308,55 @@ let eval_var_handled conf sl =
     Not_found -> Printf.sprintf " %%%s?" (String.concat "." sl)
 
 let apply_format conf nth s1 s2 =
-  let transl_nth_format s =
+  let s1  = (* perform nth selection before format check *)
     match nth with
-      Some n -> Util.ftransl_nth conf s n
-    | None -> Util.ftransl conf s
+      Some n -> Util.transl_nth conf s1 n
+    | None -> Util.transl conf s1
   in
-  match Util.check_format "%t" s1 with
-    Some s3 -> Printf.sprintf (transl_nth_format s3) (fun _ -> s2)
-  | None ->
+  try let _ = String.index s1 '%' in
```

Comment:
can you do a `if not (String.contains s1 '%') then s1 else try ...` to make it clearer